Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIBookmarks.mm

```diff
@@ -574,6 +629,45 @@ void configureGetBookmarksDelegate(TestWebExtensionManager *manager)
     [manager loadAndRun];
 }
 
+TEST_F(WKWebExtensionAPIBookmarks, BookmarksAPIRemoveAndRemoveTree)
+{
+    auto script = @[
+        @"let folder1 = await browser.bookmarks.create({ title: 'Folder One' });",
+        @"let bookmarkA = await browser.bookmarks.create({ parentId: folder1.id, title: 'Bookmark A', url: 'http://example.com/a' });",
+        @"let folder2 = await browser.bookmarks.create({ parentId: folder1.id, title: 'Folder Two' });",
+        @"let bookmarkB = await browser.bookmarks.create({ parentId: folder2.id, title: 'Bookmark B', url: 'http://example.com/b' });",
+        @"let topLevelBookmark = await browser.bookmarks.create({ title: 'Top Level Bookmark', url: 'http://example.com/top' });",
+        @"let tree1 = await browser.bookmarks.getTree();",
+        @"browser.test.assertEq(2, tree1[0].children.length, 'folder1 should have 2 children');",
+        @"browser.test.assertEq('Top Level Bookmark', tree1[1].title, 'Second top-level is topBookmark');",
+        @"browser.test.assertEq('Bookmark A', tree1[0].children[0].title, 'bookmarkA is child of folder1');",
+        @"await browser.bookmarks.remove(bookmarkA.id);",
+        @"let tree2 = await browser.bookmarks.getTree();",
+        @"browser.test.assertEq(1, tree2[0].children.length, 'folder1 should now have 1 child after removing bookmarkA');",
+        @"browser.test.assertEq('Folder Two', tree2[0].children[0].title, 'folder2 should be the only child of folder1');",
+        @"await browser.bookmarks.removeTree(folder2.id);",
+        @"let tree3 = await browser.bookmarks.getTree();",
+        @"browser.test.assertEq(0, tree3[0].children.length, 'folder1 should now have 0 children after removing folder2');",
+        @"browser.test.assertEq(2, tree3.length, 'Tree should still have 2 top-level items (folder1 and topBookmark)');",
+        @"browser.test.assertEq('Folder One', tree3[0].title, 'folder1 is still present');",
+        @"browser.test.assertEq(0, tree3[0].children.length, 'folder1 is still present but has no children');",
+        @"await browser.bookmarks.remove(folder1.id);",
+        @"let tree4 = await browser.bookmarks.getTree();",
+        @"browser.test.assertEq(1, tree4.length, 'the top level now ONLY has the top level bookmark');",
+        @"browser.test.notifyPass();",
```

Comment:
Do you want to test any errors in this test case? I'd like to see:

1) Calling `removeTree` on a leaf fails
2) Calling `remove` on a folder with children fails
3) Calling `removeTree` on an empty folder fails

You could also add tests for testing removal of a non-existent identifier.